raco/backends/myria/myria.py

```diff
@@ -1914,14 +1902,44 @@ def fire(self, op):
 
 class FillInJoinPullOrder(rules.Rule):
 
-    def __init__(self, pull_order):
+    def __init__(self):
         self._disabled = False
-        self.pull_order = pull_order
+
+    def get_idb_side(self, op):
+        for child in op.children():
+            if isinstance(child, MyriaIDBController):
+                return True
+            if self.get_idb_side(child):
+                return True
+        return False
 
     def fire(self, op):
-        if not isinstance(op, MyriaSymmetricHashJoin):
+        if not isinstance(op, algebra.UntilConvergence):
+            return op
+        if op.pull_order is None or op.pull_order == 'ALTER':
             return op
-        op.pull_order = self.pull_order
+        for ch in op.walk():
+            if isinstance(ch, MyriaSymmetricHashJoin):
+                left_idb = self.get_idb_side(ch.left)
+                right_idb = self.get_idb_side(ch.right)
+                if op.pull_order == 'PULL_IDB':
```

Comment:
As I mentioned earlier, I think this code would be more readable if `op.pull_order` became `op.pull_order_policy`.